510_Deployment/40_config.asciidoc

```diff
@@ -161,56 +134,37 @@ PUT /_cluster/settings
 }
 ----
 
-This will become a persistent setting that takes precedence over whatever is
-in the static configuration.  You should modify this setting whenever you add or
-remove master-eligible nodes.
+这将成为一个永久的配置，并且无论你配置项里配置的如何，这个将优先生效。当你添加和删除 master 节点的时候，你需要更改这个配置。
 
-==== Recovery Settings
+==== 集群恢复方面的配置
 
-Several settings affect the behavior of shard recovery when
-your cluster restarts.((("recovery settings")))((("configuration changes, important", "recovery settings")))  First, we need to understand what happens if nothing is
-configured.
+当你集群重启时，几个配置项影响你的分片恢复的表现。((("recovery settings")))((("configuration changes, important", "recovery settings")))首先，我们需要明白
+如果什么也没配置将会发生什么。
 
-Imagine you have ten nodes, and each node holds a single shard--either a primary
-or a replica--in a 5 primary / 1 replica index.  You take your
-entire cluster offline for maintenance (installing new drives, for example).  When you
-restart your cluster, it just so happens that five nodes come online before
-the other five.
+想象一下假设你有 10 个节点，每个节点只保存一个分片，这个分片是一个主分片或者是一个分片副本，或者说有一个有 5 个主分片／1 个分片副本的索引。有时你需要为整个集群做离线维护（比如，为了安装一个新的驱动程序），
+当你重启你的集群，恰巧出现了 5 个节点已经启动，还有 5 个还没启动的场景。
 
-Maybe the switch to the other five is being flaky, and they didn't
-receive the restart command right away.  Whatever the reason, you have five nodes
-online.  These five nodes will gossip with each other, elect a master, and form a
-cluster.  They notice that data is no longer evenly distributed, since five
-nodes are missing from the cluster, and immediately start replicating new
-shards between each other.
+假设其它 5 个节点出问题，或者他们根本没有收到立即重启的命令。不管什么原因，你有 5 个节点在线上，这五个节点会相互通信，选出一个 master，从而形成一个集群。
+他们注意到数据不再均匀分布，因为有 5 个节点在集群中丢失了，所以他们之间会立马启动分片复制。
 
-Finally, your other five nodes turn on and join the cluster.  These nodes see
-that _their_ data is being replicated to other nodes, so they delete their local
-data (since it is now redundant, and may be outdated).  Then the cluster starts
-to rebalance even more, since the cluster size just went from five to ten.
+最后，你的其它 5 个节点打开加入了集群。这些节点会发现 _它们_ 的数据正在被复制到其他节点，（因为这份数据要么是多余的，要么是过时的）。
+然后整个集群重新进行平衡，因为集群的大小已经从 5 变成了 10。
```

Comment:
我觉得这里的意思是集群里要发生的重平衡会更加剧烈了，所以说成『变得更加平衡』是不对的！

在 2016年3月11日 上午10:55，pengqiuyuan <notifications@github.com>写道：

> In 510_Deployment/40_config.asciidoc
> <https://github.com/elasticsearch-cn/elasticsearch-definitive-guide/pull/26#discussion_r55784061>
> :
>
> >
> > -Finally, your other five nodes turn on and join the cluster.  These nodes see
> > -that _their_ data is being replicated to other nodes, so they delete their local
> > -data (since it is now redundant, and may be outdated).  Then the cluster starts
> > -to rebalance even more, since the cluster size just went from five to ten.
> > +最后，你的其它 5 个节点打开加入了集群。这些节点会发现 _它们_ 的数据正在被复制到其他节点，（因为这份数据要么是多余的，要么是过时的）。
> > +然后整个集群重新进行平衡，因为集群的大小已经从 5 变成了 10。
>
> 根据上文的意思来看，这里翻译成 “然后集群重新进行平衡” 或者翻译成 “然后集群开始重新平衡甚至更多” 会不会翻译成 “集群变得更加平衡了”
> 要好一点儿。。因为后面一句好像强调的是“just went from” ，集群的大小刚刚从5变成了10 。 @biyuhao
> <https://github.com/biyuhao> 帮我琢磨下 😄
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elasticsearch-cn/elasticsearch-definitive-guide/pull/26/files#r55784061>
> .
>